vsphere/panels.libsonnet

```diff
@@ -0,0 +1,1118 @@
+        ])
+        + barGauge.panelOptions.withDescription('The available CPU capacity of the cluster.')
+        + barGauge.options.withOrientation('horizontal')
+        + barGauge.standardOptions.withUnit('rotmhz')
+        + barGauge.standardOptions.color.withMode('continuous-YlRd'),
+
+      clusterCPUUtilization:
+        commonlib.panels.generic.timeSeries.base.new(
+          'Cluster CPU utilization',
+          targets=[t.clusterCPUUtilization],
+          description='The CPU utilization percentage of the cluster.'
+        )
+        + g.panel.timeSeries.standardOptions.withUnit('percent')
+        + g.panel.timeSeries.options.legend.withDisplayMode('table'),
+
+      clusterMemoryEffective:
+        barGauge.new(title='Cluster memory effective')
+        + barGauge.queryOptions.withTargets([
+          t.clusterMemoryEffective,
+        ])
+        + barGauge.panelOptions.withDescription('The effective memory capacity of the cluster,')
+        + barGauge.options.withOrientation('horizontal')
+        + barGauge.standardOptions.withUnit('decbytes')
+        + barGauge.standardOptions.color.withMode('continuous-YlRd'),
+
+      clusterMemoryLimit:
+        barGauge.new(title='Cluster memory limit')
+        + barGauge.queryOptions.withTargets([
+          t.clusterMemoryLimit,
+        ])
+        + barGauge.panelOptions.withDescription('The available memory capacity of the cluster.')
+        + barGauge.options.withOrientation('horizontal')
+        + barGauge.standardOptions.withUnit('decbytes')
+        + barGauge.standardOptions.color.withMode('continuous-YlRd'),
+
+      clusterMemoryUtilization:
+        commonlib.panels.generic.timeSeries.base.new(
+          'Cluster memory utilization',
+          targets=[t.clusterMemoryUtilization],
+          description='The memory utilization percentage of the cluster.'
+        )
+        + g.panel.timeSeries.standardOptions.withUnit('percent')
+        + g.panel.timeSeries.options.legend.withDisplayMode('table'),
+
+      esxiHostsTable:
+        commonlib.panels.generic.table.base.new(
+          'ESXi hosts table',
+          targets=[
+            t.hostCPUUtilizationCluster + g.query.prometheus.withFormat('table')
+            + g.query.prometheus.withRange(true)
+            ,
+            t.hostMemoryUtilizationCluster + g.query.prometheus.withFormat('table')
+            + g.query.prometheus.withRange(true)
+            ,
+            t.networkThroughputRateCluster + g.query.prometheus.withFormat('table')
+            + g.query.prometheus.withRange(true)
+            ,
+            t.hostCPUUsageCluster + g.query.prometheus.withFormat('table')
+            + g.query.prometheus.withRange(true)
+            ,
+            t.hostMemoryUsageCluster + g.query.prometheus.withFormat('table')
+            + g.query.prometheus.withRange(true),
+          ]
+          ,
+          description='A table displaying information about the ESXi hosts in the vCenter environment.'
```

Comment:
Nit picking, but I don't think it's necessary to include the panel type in the panel description. (Applies to all table panels)